mffpy/xml_files.py

```diff
@@ -922,7 +924,10 @@ def _parse_segment(self, seg_el):
         Contains several elements <faults/>, <beginTime/> etc."""
         ret = {'status': seg_el.get('status', None)}
         for el_key, converter in self._segment_converter.items():
-            ret[el_key] = converter(self.find(el_key, seg_el))
+            el_cont = converter(self.find(el_key, seg_el))
+            # We only add the element if present in XML file
+            if el_cont is not None:
+                ret[el_key] = el_cont
```

Comment:
[required] Shouldn't this be:
```python
el_cont = self.find(el_key, seg_el)
if el_cont is not None:
    ret[el_key] = converter(el_cont)
```

I guess what you're saying is that `name` is optional and we only add it if available, but that isn't true for the rest of the iterates.

If so, I would suggest adding `name` as a special case in `_parse_keys()`:
```python
for el_key, converter in self...:
   ret[el_key] = converter(self.find(el_key, seg_el))

name_el = self.find('name', seg_el)
if name_el is not None:
    ret['name'] = str(name_el)
```